src/Adapter/Product/CommandHandler/UpdateProductCategoriesHandler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use Category;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductCategoriesCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\UpdateProductCategoriesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShopException;
+use Product;
+
+/**
+ * Handles @var UpdateProductCategoriesCommand using legacy object model
+ */
+final class UpdateProductCategoriesHandler extends AbstractProductHandler implements UpdateProductCategoriesHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateProductCategoriesCommand $command): void
+    {
+        $product = $this->getProduct($command->getProductId());
+        $defaultCategoryId = $command->getDefaultCategoryId()->getValue();
+        $categoryIds = $this->formatCategoryIdsList($command);
+
+        $this->updateCategories($product, $categoryIds);
+
+        $product->id_category_default = $defaultCategoryId;
```

Comment:
Why not use the partial update here? With `$product->setFieldsToUpdate` like in other handlers?